### PR TITLE
Add Undone transaction Command

### PR DESCRIPTION
**Pull request by ongkc**
Try various ways to retrieve person from the list using name as input for transaction constructor, but all got exception errors when i try to run. Will try to solve it by tmr evening. If you guys can solve it update me.

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -206,7 +206,6 @@ public void handleExitAppRequestEvent(ExitAppRequestEvent event) {
         this.stop();
     }
 
-    public static void main(String[] args) {
-        launch(args);
+    public static void main(String[] args) { launch(args);
```
Comment (phmignot) [inline src/main/java/seedu/address/MainApp.java]: You create a Checkstyle issue x)

File: src/main/java/seedu/address/model/transaction/Transaction.java [hunk 2]
```diff
@@ -9,19 +11,19 @@
 
 public class Transaction {
 
-    private final Person payer;
-    private final Double amount;
-    private final String description;
+    private final UniquePersonList payer;
```
Comment (phmignot) [inline src/main/java/seedu/address/model/transaction/Transaction.java]: Why did you change the type of payer into a UniquePersonList ?

File: src/main/java/seedu/address/model/person/Amount.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String MESSAGE_AMOUNT_CONSTRAINTS =
+            "Amount value can only contain numbers and comma when value is below 1";
+    public static final String AMOUNT_VALIDATION_REGEX = "^-?\\d*\\.\\d{2}$\n";
+
+    public final String value;
```
Comment (phmignot) [inline src/main/java/seedu/address/model/person/Amount.java]: Why storing the value in amount in String, instead of a double ? We are going to make some computations with these amount later.

File: src/main/java/seedu/address/model/person/Amount.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Payer paid amount in the address book.
+ */
+public class Amount {
+
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS =
+            "Payer amount can only take in numerical number with 2 decimals, and it should not be blank";
+
```
Comment (phmignot) [inline src/main/java/seedu/address/model/person/Amount.java]: "positive number"

File: src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_AMOUNT, PREFIX_DESCRIPTION, PREFIX_NAME)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTransactionCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            Name payerName = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME)).get();
+            Amount amount = ParserUtil.parseAmount(argMultimap.getValue(PREFIX_AMOUNT)).get();
+            Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION)).get();
+            Name payeeName= ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME)).get();
```
Comment (phmignot) [inline src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java]: This gets only one name of the payee. You should check how the code deals with tags (multi-optional input)

File: src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+            UniquePersonList payers = new UniquePersonList();
+            UniquePersonList payees = new UniquePersonList();
```
Comment (phmignot) [inline src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java]: And I guess it is where you are stuck, to retrieve Person from Name, to generate your list

Comment (ongkc) [reply src/main/java/seedu/address/logic/parser/AddTransactionCommandParser.java]: Yes, i will ask my friend who took the module before about how we can retrieve the person using the name.
